torch/csrc/jit/frontend/lexer.h

```diff
@@ -106,7 +106,10 @@ namespace jit {
   _(TK_DELETE, "del", "del")                     \
   _(TK_PASS, "pass", "pass")                     \
   _(TK_CLASS_DEF, "class", "class")              \
-  _(TK_IMPORT, "import", "import")
+  _(TK_IMPORT, "import", "import")               \
+  _(TK_WITH, "with", "with")                     \
+  _(TK_WITH_ITEM, "withitem", "")                \
+  _(TK_WITH_ITEM_AS, "withitemas", "as")
```

Comment:
I don't know if we ever want to support importing in TorchScript or catching exceptions.. if we do `as` can also be used as `import x as y` + `catch Exception as exc`. maybe rename `TK_WITH_ITEM_AS` to be less with statement specific?